test/framework/clusterctl/e2e_config.go

```diff
@@ -63,6 +64,19 @@ func LoadE2EConfig(ctx context.Context, input LoadE2EConfigInput) *E2EConfig {
 	return config
 }
 
+// SetCNIEnvVar read CNI from cniManifestPath and sets an environmental variable that keeps CNI resources.
+// A ClusterResourceSet can be used to apply CNI using this environmental variable.
+func SetCNIEnvVar(cniManifestPath string, cniEnvVar string) {
+	cniData, err := ioutil.ReadFile(cniManifestPath)
+	Expect(err).ToNot(HaveOccurred(), "Failed to read the e2e test CNI file")
+	Expect(cniData).ToNot(BeEmpty(), "CNI file should not be empty")
+	data := map[string]interface{}{}
+	data["resources"] = string(cniData)
+	marshalledData, err := json.Marshal(data)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(os.Setenv(cniEnvVar, string(marshalledData))).NotTo(HaveOccurred())
```

Comment:
Personally I find it symmetric to have CNI being a path and CNI_RESOURCES loaded from that path, but I'm ok in adding a prefix to it as well.
WRT to the env var, there is no need of clean up because it applies only to the scope of the running program, not to the user env.